### PR TITLE
Export folders

### PR DESCRIPTION
**Pull request by acz13**
I was sleep drunk when I slapped this together so it probably needs some cleaning up but it works...

not that I didn't add import to folder because you can just append it to restore all...

Folder view

Export view

### CODE REVIEW

File: js/storage.js [hunk 1]
```diff
@@ -1117,6 +1117,18 @@ Storage.exportAllTeams = function () {
 	}
 	return buf;
 };
+Storage.exportFolder = function (folder) {
+	var buf = '';
+	for (var i = 0, len = Storage.teams.length; i < len; i++) {
+		var team = Storage.teams[i];
+		if ((folder.indexOf("/") && team.folder === folder.slice(0, -1)) || team.format === folder) {
```
Comment (Zarel) [inline js/storage.js]: `folder.indexOf("/")` and `folder.slice(0, -1)` are weird things to do in an inner loop.

Not to mention `folder.indexOf("/")` should probably be `folder.slice(-1) === '/'`

Comment (acz13) [inline js/storage.js]: do you have a better solution?

i thought to use `folder.indexOf("/")` because it's easier lol

**Comment by acz13** [comment]
@Zarel is this good now?

cleaned up messy if statement and squashed commits

File: js/client-teambuilder.js [hunk 2]
```diff
@@ -404,6 +409,8 @@
 
 			if (window.nodewebkit) {
 				buf += '<button name="revealFolder" class="button"><i class="fa fa-folder-open"></i> Reveal teams folder</button> <button name="reloadTeamsFolder" class="button"><i class="fa fa-refresh"></i> Reload teams files</button> <button name="backup" class="button"><i class="fa fa-upload"></i> Backup/Restore all teams</button>';
+			} else if (this.curFolder) {
+				buf += '<button name="backup" class="button"><i class="fa fa-upload"></i> Backup all teams from this folder</button>';
```
Comment (Zarel) [inline js/client-teambuilder.js]: Maybe still have a `Backup all teams` button under it or something?

Comment (Zarel) [inline js/client-teambuilder.js]: I guess that would be ambiguous. This is fine.

**Comment by Zarel** [comment]
Sounds good. Can you confirm you want it (and all your future contributions to Pokémon Showdown) released under the MIT license?

**Comment by acz13** [comment]
MIT license is good

**Comment by acz13** [comment]
I just realized that this doesn't work with the (uncategorized) or genN folders... should I remove the button from those or add more code to make it work. I don't see much reason to export those anyways...

**Comment by Zarel** [comment]
I'd say remove the button from those.

**Pull request merged**